sklearn/utils/tests/test_extmath.py

```diff
@@ -457,6 +458,16 @@ def test_incremental_variance_ddof():
                                 calculated_variances, 6)
             assert_equal(incremental_count, sample_count)
 
+
+def test_eigen_sign_flip():
+    # Testing that sign flip is working & largest value has positive sign
+    random_state = np.random.RandomState(36)
+    data = random_state.randn(5, 5)
+    max_abs_rows = np.argmax(np.abs(data), axis=1)
+    data_flipped = eigen_sign_flip(data)
+    max_rows = np.argmax(data_flipped, axis=1)
+    assert_array_equal(max_abs_rows, max_rows)
```

Comment:
We should also probably test that this function does not alter the eigen data...

You could do this by manually multiplying the signs back with elements of `data_flipped` and assert that `data` and data reflipped are same :)

```
signs = np.sign(data[range(data.shape[0]), max_abs_rows])
assert_array_equal(data, data_flipped * signs[:, np.newaxis])
```